### PR TITLE
DM-36449: Add check to test is var is there before deleting

### PR DESCRIPTION
**Pull request by mfisherlevine**

### CODE REVIEW

File: tests/test_butlerUtils.py [hunk 1]
```diff
@@ -376,7 +376,8 @@ def test_dafButlerRaiseTypes(self):
         # If DAF_BUTLER_REPOSITORY_INDEX is not set *at all* then
         # using an instrument label raises a FileNotFoundError
         with unittest.mock.patch.dict('os.environ'):
-            del os.environ['DAF_BUTLER_REPOSITORY_INDEX']
+            if 'DAF_BUTLER_REPOSITORY_INDEX' in os.environ:  # can't del unless it's already there
```
Comment (ktlim) [inline tests/test_butlerUtils.py:379]: It's more compact to say `os.environ.pop('DAF_BUTLER_REPOSITORY_INDEX', None)`, but it might be less clear.

**Review by ktlim — APPROVED** [review]
This change alone is fine.  I haven't looked at any of the other code, though, and problems remain with packages in lsst_sitcom (e.g. missing summit_extras).

**Comment by mfisherlevine** [comment]
> This change alone is fine. I haven't looked at any of the other code, though, and problems remain with packages in lsst_sitcom (e.g. missing summit_extras).

Yup, woke up to that fail on Jenkins. Should just be the fact that `summit_extras` wasn't in the table file for `rubintv_production` so I've added that to this same ticket number and am rerunning now...

File: tests/test_butlerUtils.py (hunk 1 above)
Comment (mfisherlevine) [reply tests/test_butlerUtils.py:379]: Yeah, I think I like the explicit way in this instance.

**Pull request merged**